### PR TITLE
hack/env: Remove tmp volume

### PR DESCRIPTION
**Pull request by mfojtik**
- Use host TMPDIR or pass in /tmp and /var/tmp from host instead of
  using a volume for /tmp
- Add BASEVARTMPDIR variable and update tito to use it

### CODE REVIEW

**Comment by mfojtik** [comment]
rebased version of https://github.com/openshift/origin/pull/16686

+ fixes building on OSX

@detiber @stevekuznetsov can you guys double check this? there were several conflicts as the hack scripts moved, so I hope I haven't missed anything.

File: hack/lib/build/environment.sh [hunk 1]
```diff
@@ -15,6 +15,37 @@ function os::build::environment::create() {
   workingdir=$( os::build::environment::release::workingdir )
   additional_context+=" -w ${workingdir}"
 
+  local tmpdir_template="openshift-env-XXXXXXXXXX"
+  local platform=$( os::build::host_platform_friendly )
+  # The mktemp --tmpdir does not work on OSX and /var/tmp/.. extra steps to
+  # share with Docker VM
+  if [[ $platform != "mac" ]]; then
```
Comment (mfojtik) [inline hack/lib/build/environment.sh]: @detiber @stevekuznetsov i don't think we care about this on mac?

**Comment by mfojtik** [comment]
/retest

aws flake

File: hack/build-rpm-release.sh [hunk 2]
```diff
@@ -1 +0,0 @@
-build-rpms.sh
```
Comment (soltysh) [inline hack/build-rpm-release.sh]: Why did you change that? This supposed to be link to `hack/build-rpms.sh`?

Comment (soltysh) [reply hack/build-rpm-release.sh]: Even more the two are not different after your pr.

Comment (mfojtik) [reply hack/build-rpm-release.sh]: accident during rebase

**Comment by mfojtik** [comment]
@detiber @stevekuznetsov wonder if tmpfs is what we want for things other than etcd:

# github.com/openshift/origin/tools/genman
/usr/lib/golang/pkg/tool/linux_amd64/link: flushing /tmp/go-link-665964831/go.o: write /tmp/go-link-665964831/go.o: no space left on device

**Comment by stevekuznetsov** [comment]
I thought the build was supposed to go into `/var/tmp`? IIRC the conversation is in the previous thread -- builds in `/tmp` are not going to work for the reason you already saw

**Comment by stevekuznetsov** [comment]
This seems like too bold of an approach, fixed this another way.

/close
